alpha/apps/kaltura/lib/kUploadTokenMgr.php

```diff
@@ -176,6 +176,19 @@ public function uploadFileToToken($fileData, $resume = false, $resumeAt = -1)
 		$this->_uploadToken->setDc(kDataCenterMgr::getCurrentDcId());
 		
 		$this->_uploadToken->save();
+		
+		$flavorAsset = assetPeer::retrieveById($this->_uploadToken->getObjectId());
```

Comment:
this logic needs to run only if the upload was finished fully and the flavor asset is source and $this->_uploadToken->getObjectId()